apps/server/src/shared/infra/preview-generator/preview.producer.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { AmqpConnection } from '@golevelup/nestjs-rabbitmq';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { ConfigService } from '@nestjs/config';
+import { Test, TestingModule } from '@nestjs/testing';
+import { setupEntities } from '@shared/testing';
+import { LegacyLogger } from '@src/core/logger';
+import { ErrorMapper, FilesPreviewEvents, FilesPreviewExchange } from '../rabbitmq';
+import { PreviewFileOptions } from './interface';
+import { PreviewProducer } from './preview.producer';
+
+describe('PreviewProducer', () => {
+	let module: TestingModule;
+	let service: PreviewProducer;
+	let configService: DeepMocked<ConfigService>;
+	let amqpConnection: DeepMocked<AmqpConnection>;
+
+	const timeout = 10000;
+
+	beforeAll(async () => {
+		await setupEntities();
+		module = await Test.createTestingModule({
+			providers: [
+				PreviewProducer,
+				{
+					provide: LegacyLogger,
+					useValue: createMock<LegacyLogger>(),
+				},
+				{
+					provide: AmqpConnection,
+					useValue: createMock<AmqpConnection>(),
+				},
+				{
+					provide: ConfigService,
+					useValue: createMock<ConfigService>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(PreviewProducer);
+		amqpConnection = module.get(AmqpConnection);
+		configService = module.get(ConfigService);
+		configService.get.mockReturnValue(timeout);
```

Comment:
Kannst du bitte configService.get.mockReturnValue(timeout); in die setups verschieben wo es benötigt wird, evtl. auch einfach direkt die 10000 eintragen, ohne über const timeout gehen.